virtcontainers/network.go

```diff
@@ -234,88 +234,59 @@ func generateEndpoints(typedEndpoints []TypedJSONEndpoint) ([]Endpoint, error) {
 	var endpoints []Endpoint
 
 	for _, e := range typedEndpoints {
+		var (
+			endpointInf Endpoint
+			err         error
+		)
 		switch e.Type {
 		case PhysicalEndpointType:
 			var endpoint PhysicalEndpoint
-			err := json.Unmarshal(e.Data, &endpoint)
-			if err != nil {
-				return nil, err
-			}
-
-			endpoints = append(endpoints, &endpoint)
-			networkLogger().WithFields(logrus.Fields{
-				"endpoint":      endpoint,
-				"endpoint-type": "physical",
-			}).Info("endpoint unmarshalled")
+			err = json.Unmarshal(e.Data, &endpoint)
```

Comment:
looks good. Can we take out these two lines out of every switch case as well and add it at the end of the switch?

Reply:
They are all different types, I'll try to see if interface can satisfy this.